src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs

```diff
@@ -580,6 +592,116 @@ public async Task TestNuGetAndVsixAnalyzer_MultipleNuGetAnalyzersCollectivelyRep
+                vsixAnalyzers: ImmutableArray.Create(vsixAnalyzer),
+                expectedVsixAnalyzersExecuted: true,
+                nugetSuppressors: ImmutableArray<NuGetSuppressor>.Empty,
+                expectedNugetSuppressorsExecuted: false,
+                vsixSuppressors: ImmutableArray.Create(vsixSuppressor),
+                expectedVsixSuppressorsExecuted: true);
+
+            // All without overlap, the VSIX analyzer and suppressor still work when nuget analyzers are present:
+            //   1) No duplicate diagnostics
+            //   2) All analyzers execute
+            //   3) VSIX diagnostics are suppressed.
+            await TestNuGetAndVsixAnalyzerCoreAsync(
+                nugetAnalyzers: ImmutableArray.Create(firstNugetAnalyzer, secondNugetAnalyzer),
+                expectedNugetAnalyzersExecuted: true,
+                vsixAnalyzers: ImmutableArray.Create(vsixAnalyzer),
+                expectedVsixAnalyzersExecuted: true,
+                nugetSuppressors: ImmutableArray<NuGetSuppressor>.Empty,
+                expectedNugetSuppressorsExecuted: false,
+                vsixSuppressors: ImmutableArray.Create(vsixSuppressor),
+                expectedVsixSuppressorsExecuted: true,
+                new[]
+                {
+                    (Diagnostic("A", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
+                    (Diagnostic("B", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
+                    (Diagnostic("C", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer))
+                });
+
+            // All without overlap, verify the following:
+            //   1) No duplicate diagnostics
+            //   2) Both NuGet and Vsix analyzers execute
+            //   3) Appropriate diagnostic filtering is done - Nuget suppressor suppresses VSIX analyzer.
+            await TestNuGetAndVsixAnalyzerCoreAsync(
+                nugetAnalyzers: ImmutableArray.Create(firstNugetAnalyzer),
+                expectedNugetAnalyzersExecuted: true,
+                vsixAnalyzers: ImmutableArray.Create(vsixAnalyzer),
+                expectedVsixAnalyzersExecuted: true,
+                nugetSuppressors: ImmutableArray.Create(nugetSuppressor),
+                expectedNugetSuppressorsExecuted: true,
+                vsixSuppressors: ImmutableArray<VsixSuppressor>.Empty,
+                expectedVsixSuppressorsExecuted: false,
+                new[]
+                {
+                    (Diagnostic("A", "Class").WithLocation(1, 7), nameof(NuGetAnalyzer)),
+                });
+
+            // Suppressors with duplicate support for VsixAnalzer, but not 100% overlap. Verify the following:
+            //   1) No duplicate diagnostics
+            //   2) Both NuGet and Vsix analyzers execute
+            //   3) Both Nuget and Vsix suppressors execute
+            //   4) Appropriate diagnostic filtering is done - Nuget suppressor and Vsix suppressor together suppresses VSIX analyzer.
```

Comment:
Let's simplify this:
```suggestion
            // Suppressors with duplicate support for VsixAnalzer, but not 100% overlap. Verify the following:
            //   1) No duplicate diagnostics
            //   2) Both NuGet and Vsix analyzers execute
            //   3) Only Nuget suppressors execute
            //   4) Appropriate diagnostic filtering is done - Nuget suppressor suppresses VSIX analyzer.
```

Reply:
Per offline discussion, going with this approach.